monte_carlo.ipynb

```diff
@@ -183,62 +190,36 @@
+    "pricesCompaniesAdjustments = np.array([[[[[pricesCompanies[k,a,w]  for k in range(numberPlayers)]for _ in range(numberPriceAdjustmentsBeforeHorizon[s])] for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "pricesCompaniesAdjustments = np.array([[[np.array(calculatePrices(pricesCompaniesAdjustments[a,w,s], adjustingFirm[s])) for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "\n",
+    "gen = np.array([[[[list((x for x in range(numberCustomerEvents) if timeMarketEvents[s][i]< timeCustomerEvents[s][x]< timeMarketEvents[s][i+1])) for i in range(numberPriceAdjustmentsBeforeHorizon[s])] for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "randomCustomerScoring = np.array([[[[[[(pricesCompaniesAdjustments[a,w,s][i,k] + customerScoringQualityRating[w,s,c,k]) for k in range(numberPlayers)]for c in gen[a,w,s][i]]for i in range(numberPriceAdjustmentsBeforeHorizon[s])] for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "randomCustomerDecision = np.array([[[[[np.argmin(randomCustomerScoring[a,w,s][i][c]) if min(randomCustomerScoring[a,w,s][i][c])< random.uniform(5,15) else -1 for c in range(len(gen[a,w,s][i]))]for i in range(numberPriceAdjustmentsBeforeHorizon[s])] for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "As = np.array([[[[np.count_nonzero(np.hstack(randomCustomerDecision[a,w,s])== k) for k in range(numberPlayers)]for s in range(numberSimulations)] for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "\n",
+    "A = np.array([[[[1/numberSimulations*np.count_nonzero(As[a,w,:,k]==x) for k in range(numberPlayers)]for x in range(6)]for w in range(numberConsideredMarketSituations)] for a in range(len(prices))])\n",
+    "P_ori = [[1- A[a,w,0,watchedCompany]for w in range(numberConsideredMarketSituations)] for a in range(len(prices))]\n",
+    "P_ori2 = [[sum([A[a,w,2,watchedCompany]for x in range(2,6)])for w in range(numberConsideredMarketSituations)] for a in range(len(prices))]\n",
+    "\n",
+    "EW_ori = [[prices[a]*P_ori[a][w] for w in range(numberConsideredMarketSituations)]for a in range(len(prices))]\n",
     "EW_est = [[prices[a]*predictionEstimated[w][a] for w in range(numberConsideredMarketSituations)]for a in range(len(prices))]\n",
     "\n",
-    "D_P = [1/numberConsideredMarketSituations*sum([abs(P_ori[w][a]- predictionEstimated[w][a]) for w in range(numberConsideredMarketSituations)]) for a in range(len(prices))]\n",
+    "D_P = [1/numberConsideredMarketSituations*sum([abs(P_ori[a][w]- predictionEstimated[w][a]) for w in range(numberConsideredMarketSituations)]) for a in range(len(prices))]\n",
```

Comment:
Just want to be sure: this change was intentional, yes?

Reply:
Yes, the change was intentional.